src/omotes_orchestrator/postgres_job_manager.py

```diff
@@ -0,0 +1,101 @@
+import logging
+from datetime import datetime, timezone
+import time
+import threading
+from omotes_orchestrator.config import PostgresJobManagerConfig
+from omotes_orchestrator.postgres_interface import PostgresInterface
+from omotes_orchestrator.db_models.job import JobDB
+
+
+LOGGER = logging.getLogger("omotes_orchestrator")
+
+
+class PostgresJobManager:
+    """Periodically checks the job row in the database and cleans up the stale one."""
+
+    postgresql_if: PostgresInterface
+    """Interface to PostgreSQL."""
+    postgres_job_manager_config: PostgresJobManagerConfig
+    """PostgresJobManager configuration"""
+    _init_time: datetime
+    """Instantiated datetime (UTC), is typically instantiated when the orchestrator is started"""
+    _active_threshold_sec: int
+    """Only when the instance is initialized longer than the threshold period
+    can it start cleaning the stale jobs."""
+    _job_retention_sec: int
+    """The allowed retention time in seconds of a postgres job row"""
+    _rerun_sec: int
+    """The period in seconds to rerun the stale job cleaning task"""
+
+    def __init__(self,
+                 postgresql_if: PostgresInterface,
+                 postgres_job_manager_config: PostgresJobManagerConfig) -> None:
+        """Construct the postgres job manager."""
+        self.postgresql_if = postgresql_if
+        self.postgres_job_manager_config = postgres_job_manager_config
+        self._init_time = datetime.now(timezone.utc)
+        self._active_threshold_sec = self.postgres_job_manager_config.job_retention_sec
```

Comment:
Should this get the same value as `self._job_retention_sec`?
See comment below as well.

Reply:
As discussed, no action for now.